src/js/controllers/ctrl-create.js

```diff
@@ -542,7 +556,9 @@ ${msg.toLowerCase()}`,
 		getQset().then(() => {
 			if (!$scope.invalid) {
 				$q(resolve => resolve(inst_id))
+					.then(widgetSrv.lockWidget)
 					.then(widgetSrv.getWidget)
```

Comment:
I'm thinking we should try to do lockWidget, getWidget, and checkUserPublishPerms in parallel?

Reply:
I'm not sure - if the widget is locked then we shouldn't try to get it, and if we can't get it then we shouldn't bother checking whether the user can publish it. That, and `checkUserPublishPerms` relies on the full widget data being passed into it from `widgetSrv.getWidget`.